docs/tutorial/4-code-update-repeat.md

```diff
@@ -0,0 +1,167 @@
+
+ * `context`: build context for image build (specified here as the current directory, which is the repo root)
+
+   Tilt watches for changes to any modified files in this directory or any subdirectory, recursively.
+   Perfect! We changed `./web/vite.config.js`, which meets this criteria.
+   
+   However, let's keep looking at the other arguments to make sure they don't negate or alter this somehow... 
+
+ * `only` (optional): filters paths included in build context and restricts file watching to the subset of paths
+
+    Because we have a "mono-repo" (multiple services in a single repository) and the build context is the repo root (`.`), we set this to `['web/']` so that unrelated changes, such as those to the backend (files under `api/`) don't trigger a re-build of the "web" resource.
+    Since `./web/vite.config.js` _is_ under `./web/`, it hasn't been excluded, which is what we want!
+
+    Just one more argument to go...
+ 
+ * `ignore` (optional): excludes certain paths from the build context and ignores changes to them
+
+    We've used this to exclude `./web/dist/` for our production web assets, which otherwise match the rules defined by `context` and `only`.
+    This is supplementary to `.dockerignore`, but can be helpful for cases where you want different ignore rules for local dev with Tilt, for example.
+
+    As `./web/vite.config.js` is _not_ under `./web/dist/`, it was not ignored, which is what we'd expect.
+ 
+If we put all this together, Tilt is watching for any file changes in the `web/` directory or any of its subdirectories, recursively, EXCEPT for those in `web/dist` (or any of its subdirectories, recursively).
+Phew!
+
+When a matching file changes, such as `web/vite.config.js`, because it's watched by the `tilt-avatar-web` container image build configuration, Tilt initiates an update for the "web" resource.
+
+> **How does Tilt know the `tilt-avatar-web` image belongs to the "web" resource?**
+>
+> You might remember that a resource can be composed of multiple bits of work.
+> In the case of the "web" resource, it has a container image build and a Kubernetes Deployment.
+>
+> Tilt associated the `tilt-avatar-web` container image with the "web" resource because the container image name is referenced in Kubernetes YAML loaded in the `Tiltfile` with `k8s_yaml`.
+> (This is not the only way that container images can be assembled into a resource, and it's possible to manually configure where auto-assembly is insufficient.)
+
+
+## 2. Resource Update
+Now, the update process starts:
+```log
+STEP 1/3 — Building Dockerfile: [tilt-avatar-web]
+  ...
+
+STEP 2/3 — Pushing localhost:44099/tilt-avatar-web:tilt-0b9fcdf9cfea47ba
+  ...
+
+STEP 3/3 — Deploying
+     Injecting images into Kubernetes YAML
+     Applying via kubectl:
+     → web:deployment
+```
+First, Tilt built an updated version of the container image.
+Then, it pushed it to our local registry so that it can be used by Kubernetes.
```

Comment:
(nit) a couple `it`s in this sentence made me double-take. 
```suggestion
Then, it pushed the image to our local registry so that it can be used by Kubernetes.
```